js/script.js

```diff
@@ -6,10 +6,128 @@ const navSlide = () => {
     nav.classList.toggle('nav-active');
     burger.classList.toggle('toggle');
   });
+
   document.querySelectorAll('.nav-link').forEach((n) => n.addEventListener('click', () => {
     nav.classList.remove('nav-active');
     burger.classList.remove('toggle');
   }));
 };
 
 navSlide();
+
+/* single Project section modal */
+
+const modalBtn = document.querySelector('.project-button');
+const modalBg = document.querySelector('.modal-section');
+const modalClose = document.querySelector('.close-btn');
+
+modalBtn.addEventListener('click', () => {
+  modalBg.style.visibility = 'visible';
+  modalBg.style.opacity = 1;
+});
+modalClose.addEventListener('click', () => {
+  modalBg.style.visibility = 'hidden';
+  modalBg.style.opacity = 0;
+});
+
+/* card-modal */
+
+const popupBtn = document.querySelectorAll('.card-button');
+
+const popupTitle = [];
+document.querySelectorAll('div.flex-container > h4').forEach((item) => {
+  popupTitle.push(item.textContent);
+});
+
+const popupTechno = ['HTML', 'Bootstrap', 'Ruby on Rails'];
+
+const popupDescrip = 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the releaLorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the releax map lapora verita.';
+const popupLive = 'file:///D:/micro_projects/portfolio_mobile/index.html';
+
+const popupSource = 'https://github.com/ZuhRa-HashiMi';
+
+const popupObj = {};
+for (let i = 0; i < popupTitle.length; i += 1) {
+  popupObj[`Project${i}`] = {};
+  popupObj[`Project${i}`].title = popupTitle[i];
+  popupObj[`Project${i}`].technologies = popupTechno;
+  popupObj[`Project${i}`].description = popupDescrip;
+  popupObj[`Project${i}`].liveLink = popupLive;
+  popupObj[`Project${i}`].sourceLink = popupSource;
+}
```

Comment:
- I applaud the efforts you have put into this project 👏 👏 . But according to the project requirements, the details for all  your project should be an array of objects. For example `const projects = [{
    name: "Submarine game",
    description: "A Phaser3 platform game built as part of the Microverse JavaScript program.",
    image: "link_to_image.jpg",
    technologies: ["Phaser3", "JavaScript", "Jest"],
    link: "link_to_live_version.com",
}]`.Doing this will ensure that subsequent projects can be easily added to the array only and can easily be dynamically loaded to the DOM. Please try to fix this 👍 👍

Reply:
Thanks for feedback, but can you tell me that how can I solve this.
please tell me a little bit clearly to know.
@mckent05

Reply:
Hello @ZuhRa-HashiMi, you should have an array of objects that contains the details of all projects as I suggested in my review. Then using the `querySelectorAll` method, you can get all the 'view project' buttons in the DOM. Based on the index of the project button clicked, you can display the project details in the popup from the array as you have already done. Let me know if you need further clarification 😄 😄